File: remove_logs.py
import os
import re

src_dir = r'c:\Users\opg21\OneDrive\Desktop\Work 🍃\exam 01\src'

for root, _, files in os.walk(src_dir):
    for file in files:
        if file.endswith(('.js', '.jsx')):
            filepath = os.path.join(root, file)
            try:
                with open(filepath, 'r', encoding='utf-8') as f:
                    lines = f.readlines()
            except UnicodeDecodeError:
                with open(filepath, 'r', encoding='utf-16') as f:
                    lines = f.readlines()
            
            changed = False
            new_lines = []
            for i in range(len(lines)):
                if 'console.log' in lines[i]:
                    changed = True
                    # If it's a simple line containing just console.log(...)
                    if re.match(r'^\s*console\.log\(.*\);?\s*$', lines[i]):
                        new_lines.append(re.sub(r'console\.log\(.*\);?', '/* log removed */', lines[i]))
                    elif re.search(r'=>\s*console\.log', lines[i]):
                        new_lines.append(re.sub(r'console\.log\([^)]*\)', 'undefined', lines[i]))
                    else:
                        new_lines.append(lines[i].replace('console.log', '/* log removed */'))
                else:
                    new_lines.append(lines[i])
            
            if changed:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.writelines(new_lines)
                pass

File: remove_logs.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
  });
}

const srcDir = path.resolve('c:/Users/opg21/OneDrive/Desktop/Work 🍃/exam 01/src');

walkDir(srcDir, function(filePath) {
  if (filePath.endsWith('.js') || filePath.endsWith('.jsx')) {
    let content = fs.readFileSync(filePath, 'utf8');
    if (content.includes('console.log')) {
      // Replace console.log(...) with empty string or void 0
      // This simple regex handles basic cases. It matches console.log followed by anything up to the balancing parenthesis.
      // For a truly safe removal, it's better to just comment out the line if it's the only thing on the line
      let lines = content.split('\n');
      let changed = false;
      for (let i = 0; i < lines.length; i++) {
          if (lines[i].includes('console.log')) {
              // If the line only contains console.log(...) and maybe some spaces and semicolons
              if (/^\s*console\.log\(.*\);?\s*$/.test(lines[i])) {
                  lines[i] = lines[i].replace(/console\.log\(.*\);?/, '/* log removed */');
                  changed = true;
              } else if (/=>\s*console\.log/.test(lines[i])) {
                  lines[i] = lines[i].replace(/console\.log\(.*\)/, 'undefined');
                  changed = true;
              } else {
                  lines[i] = lines[i].replace(/console\.log/g, '/* log removed */');
                  changed = true;
              }
          }
      }
      if (changed) {
          fs.writeFileSync(filePath, lines.join('\n'));
          console.log(`Updated ${filePath}`);
      }
    }
  }
});
